pymc3/sampling.py

```diff
@@ -430,23 +438,30 @@ def init_nuts(init='advi', n_init=500000, model=None, **kwargs):
 
     pm._log.info('Initializing NUTS using {}...'.format(init))
 
+    random_seed = int(np.atleast_1d(random_seed)[0])
+
     if init == 'advi':
-        v_params = pm.variational.advi(n=n_init)
-        start = pm.variational.sample_vp(v_params, 1, progressbar=False, hide_transformed=False)[0]
+        v_params = pm.variational.advi(n=n_init, random_seed=random_seed)
+        start = pm.variational.sample_vp(v_params, njobs, progressbar=False,
+        hide_transformed=False, random_seed=random_seed)
+        if njobs == 1:
+            start = start[0]
         cov = np.power(model.dict_to_array(v_params.stds), 2)
     elif init == 'advi_map':
         start = pm.find_MAP()
-        v_params = pm.variational.advi(n=n_init, start=start)
+        v_params = pm.variational.advi(n=n_init, start=start,
+        random_seed=random_seed)
         cov = np.power(model.dict_to_array(v_params.stds), 2)
     elif init == 'map':
         start = pm.find_MAP()
         cov = pm.find_hessian(point=start)
-
     elif init == 'nuts':
-        init_trace = pm.sample(step=pm.NUTS(), draws=n_init)
-        cov = pm.trace_cov(init_trace[n_init//2:])
-
-        start = {varname: np.mean(init_trace[varname]) for varname in init_trace.varnames}
+        init_trace = pm.sample(step=pm.NUTS(), draws=n_init,
+        random_seed=random_seed)[n_init//2:]
```

Comment:
@aloctavodia once these minor issues are fixed we can start to merge this I think.